web/src/routes/(user)/map/+page.svelte

```diff
@@ -0,0 +1,90 @@
+<script lang="ts">
+	import type { PageData } from '../map/$types';
+	import UserPageLayout from '$lib/components/layouts/user-page-layout.svelte';
+	import { Map, TileLayer, AssetMarkerCluster } from '$lib/components/shared-components/leaflet';
+	import Portal from '$lib/components/shared-components/portal/portal.svelte';
+	import AssetViewer from '$lib/components/asset-viewer/asset-viewer.svelte';
+	import {
+		assetInteractionStore,
+		isViewingAssetStoreState,
+		viewingAssetStoreState
+	} from '$lib/stores/asset-interaction.store';
+	import { colorTheme } from '$lib/stores/preferences.store';
+
+	export let data: PageData;
+
+	let initialMapCenter: [number, number] = [48, 11];
+
+	if (data.mapMarkers.length) {
+		let firstMarker = data.mapMarkers[0];
+		initialMapCenter = [firstMarker.lat, firstMarker.lon];
+	}
+
+	let viewingAssets: string[] = [];
+	let viewingAssetCursor = 0;
+
+	function onViewAssets(assets: string[]) {
+		assetInteractionStore.setViewingAssetId(assets[0]);
+		viewingAssets = assets;
+		viewingAssetCursor = 0;
+	}
+
+	function navigateNext() {
+		if (viewingAssetCursor < viewingAssets.length - 1) {
+			assetInteractionStore.setViewingAssetId(viewingAssets[++viewingAssetCursor]);
+		}
+	}
+
+	function navigatePrevious() {
+		if (viewingAssetCursor > 0) {
+			assetInteractionStore.setViewingAssetId(viewingAssets[--viewingAssetCursor]);
+		}
+	}
+</script>
+
+<UserPageLayout user={data.user} title={data.meta.title}>
+	<div slot="buttons" />
+
+	<div class="h-[90%] w-full">
+		<Map latlng={initialMapCenter} zoom={7}>
```

Comment:
If I recall correctly, directly importing Leaflet (or components that import it) will lead to errors in production builds. Can you please validate that everything still works in production? I've previously [lazy loaded](https://github.com/immich-app/immich/blob/6acfb55dccf51d7a86a5f664cc9e40974cbc38dd/web/src/lib/components/asset-viewer/detail-panel.svelte#L213) all Leaflet components

Reply:
Strange, but thanks for the hint. I just copied your approach and it seems to work well.

Reply:
It happens because Leaflet accesses `window` which isn't available for server side rendered responses. Lazy loading worked previously because a user interaction would trigger the component load, so the server never attempted to render it. But now Leaflet is loaded on a page, which means the server will attempt to load this anyway and an error is logged (the page still works though).

To get rid of the logged error, I can think of two solutions:

1. Wrap the await import in a check to see if it's running in a browser.
```svelte
{#if browser}
	{#await import()}{/await}
{/if}
```

2. Remove the await import and disable SSR for the map page
```ts
// +page.server.ts
export const ssr = false;
```

Reply:
This still happens after the await fix. Loading the page doesn't error out anymore in production, but I'm still seeing an error being logged (without impacting usability) in the web container

Reply:
Weird. I don't see it in my logs locally.